core/src/main/java/com/linecorp/armeria/server/throttling/tokenbucket/BandwidthLimit.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.throttling.tokenbucket;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+
+import javax.annotation.Nonnull;
+
+import com.google.common.base.MoreObjects;
+
+import io.github.bucket4j.Bandwidth;
+import io.github.bucket4j.Refill;
+
+/**
+ * Stores configurations of a Token-Bucket bandwidth limit.
+ */
+public class BandwidthLimit {
+
+    private final long limit;
+    private final long overdraftLimit;
```

Comment:
do you want to start with overdraft support? If not, you can probably vendor the dependency as bucket limiting implementation can maybe be shared from other code if not for this.

Reply:
Overdraft is very important for our use cases. There are other important things which I did not (yet) add, like refill mode.